app/src/main/java/com/github/se/orator/ui/overview/SpeakingPracticeModule.kt

```diff
@@ -97,23 +111,113 @@ fun SpeakingPracticeModule(
 
                 // Dynamically generated input fields based on the provided data
                 inputs.forEach { input ->
-                  OutlinedTextField(
-                      value = input.value,
-                      onValueChange = input.onValueChange,
-                      label = { Text(input.label, color = AppColors.textColor) },
-                      placeholder = { Text(input.placeholder, color = AppColors.textColor) },
-                      modifier =
-                          Modifier.fillMaxWidth()
-                              .height(
-                                  input.height.dp) // Assuming 'height' is defined in InputFieldData
-                              .testTag(input.testTag),
-                      colors =
-                          TextFieldDefaults.outlinedTextFieldColors(
-                              focusedBorderColor = AppColors.primaryColor,
-                              unfocusedBorderColor = AppColors.textColor,
-                              cursorColor = AppColors.primaryColor,
-                              focusedLabelColor = AppColors.primaryColor,
-                              unfocusedLabelColor = AppColors.textColor))
+                  if (input.isDropdown) {
+                    // Handle dropdown input
+                    FocusAreaDropdown(
+                        selectedFocusArea = input.value,
+                        onFocusAreaSelected = input.onValueChange,
+                        question = input.question,
+                        placeholder = input.placeholder,
+                        testTag = input.testTag,
+                        dropdownItems = input.dropdownItems)
+                  } else {
+                    // Display question as Text
+                    Text(
+                        text = input.question,
+                        style = AppTypography.smallTitleStyle.copy(fontWeight = FontWeight.Bold),
+                        color = AppColors.textColor,
```

Comment:
The colors used should come from MaterialTheme.colorScheme to be linked with the centralized color manager. This issue will potentially be fixed once you merge this branch with the most recent version of the main. If not then you should changed it. This comment is valid for all the occurrences of  the usage of AppColors.

Reply:
Done!